### PR TITLE
Gradient clipping component

### PR DESCRIPTION
**Pull request by le1nux**
* additionally refactored the config into pedantic interface types, component instantiation models and component config

### CODE REVIEW

File: src/modalities/trainer.py [hunk 1]
```diff
@@ -40,23 +42,25 @@ def __init__(
...
         loss_fun: Loss,
         train_step_id: int,
         data_loader: LLMDataLoader,
     ) -> Tuple[torch.Tensor, torch.Tensor]:
         result_batch = model_predict_batch(model=model, batch=batch)
-        loss = loss_fun(result_batch) / self.gradient_acc_steps
-        loss.backward()
-        gradient_norm_score = self.gradient_clipper(model)
+        loss = loss_fun(result_batch)
+        (loss / self.gradient_acc_steps).backward()
```
Comment (le1nux) [inline src/modalities/trainer.py:54]: tbd

File: src/modalities/trainer.py [hunk 2]
```diff
@@ -98,9 +103,13 @@ def train(
             forward_backward_time_recorder.stop()
             # Save the batch loss
             cumulated_loss_and_gradient_norm[0] += batch_loss.item()
-            cumulated_loss_and_gradient_norm[1] += gradient_norm_score.item()
             # This works, because we always drop the last batch in case it has less samples than the batch size
             cumulated_loss_and_gradient_norm[-1] += 1  # number of local batches
```
Comment (le1nux) [inline src/modalities/trainer.py:107]: rename to cumulated_loss

File: src/modalities/trainer.py [hunk 3]
```diff
@@ -181,7 +192,8 @@ def train(
 
     def _reset_loss_and_gradient_norm(self):
```
Comment (le1nux) [inline src/modalities/trainer.py:193]: rename

**Review by mali-git — APPROVED** [review]
LGTM besides the discussed comments.
